js/src/collapse.js

```diff
@@ -242,7 +242,14 @@ class Collapse extends BaseComponent {
 
     this.setTransitioning(true)
 
-    const complete = () => {
+    const complete = event => {
+      // if event is dispatched by a nested element reschedule the event handler
+      if (event.target !== this._element) {
+        EventHandler.one(this._element, 'transitionend', complete)
+        emulateTransitionEnd(this._element, transitionDuration)
```

Comment:
From what I can tell, it should work without `emulateTransitionEnd` but the test did fail in that case. Either we leave this here or dispatch the event manually in the spec. Not sure what is preferred.

Reply:
If it's only for our tests, we should handle that there. But only if we are 100% sure it affects the tests only :)

Reply:
@XhmikosR appreciate your response. That made me think a little more about this. :+1:

So here is the thing. If I understand the purpose of `emulateTransitionEnd()` correctly, it's some kind of fallback for browsers that don't support `transitionend` events natively. Since it uses the `triggerTransitionEnd()` function, which dispatches events that _don't_ bubble it shouldn't be possible to reach that block in those browsers.

IMO `event.taget !== this._element` can only be true if the event is coming from a nested element. In that case calling this function is kind of irrelevant, because if it's a bubbling event it's likely because there is native support.

But there is more to `emulateTransitionEnd()`. It will queue a callback (that dispatches a custom `transitionend` event) that is called after the transition-duration, _in case_ there is no other `transitionend` event that would prevent that. Although it doesn't care if the `transitionend` event came from a nested element. (Maybe it makes sense to change it there but this could introduce problems all over the place. 🤔)

To summarize the behavior:

In a browser that supports `transitionend` natively, there will be two events dispatched by the browser. One from the input (that bubbles) and another from the collapse. The event from the input will be handled by the if condition and the event from the collapse will call through. `emulateTransitionEnd()` won't do anything here because it's cancelled by the collapse's event so it wouldn't be necessary to put it there IMO.

In a browser that doesn't support `transitionend`, there will be no event for the input but only for the collapse dispatched by `emulateTransitionEnd()`. So the if condition is never met and `complete` will just go on.

In our unit tests, we don't rely on the transitions in our CSS but only on those triggered by JS. In this specific case I'm manually dispatching the event (that bubbles like the native ones) from the input in the spec. This prevents `emulateTransitionEnd()` from dispatching a custom `transitionend`, so `complete` would be called only once by the event from the input. This will go through the if and since I'm calling `emulateTransitionEnd()` within the if block it's queuing another `transitionend` event that will call `complete` a second time. This could be done in the spec instead.

So..
- Either I will change `emulateTransitionEnd()` to ignore events from nested elements which preventing it from dispatching the custom `transitionend`.
- Or remove `emulateTransitionEnd()` from the if block and dispatch the event manually in the spec.
- Or leave it like it is.

Your choice 😊